src/lib/optimizer/join_ordering/join_graph.cpp

```diff
@@ -0,0 +1,139 @@
+#include "join_graph.hpp"
+
+#include <memory>
+#include <optional>
+#include <utility>
+#include <vector>
+
+#include "constant_mappings.hpp"
+#include "join_edge.hpp"
+#include "join_plan_predicate.hpp"
+#include "types.hpp"
+#include "utils/assert.hpp"
+
+namespace {
+
+using namespace opossum;  // NOLINT
```

Comment:
?

Reply:
lint doesn't allow using namespace and I refuse to prefix everything with `opossum::` just because I want to use an anonymous namespace.

Reply:
What do you need an anonymous namespace for, anyway? You are in a cpp file, so your names won't leak.

Reply:
I'm pretty sure they will?!

Reply:
They will "leak" as in they become a linker symbol and we don't wanna risk duplicates there. That's the point of anonymous namespaces, is it not?

Reply:
Looks like static vs namespace is not trivial:
https://stackoverflow.com/questions/154469/unnamed-anonymous-namespaces-vs-static-functions?noredirect=1&lq=1